node/middlewares/newSitemap.ts

```diff
@@ -0,0 +1,91 @@
+import * as cheerio from 'cheerio'
+import { keys, map, mergeAll, prop, range, replace } from 'ramda'
+
+import { Internal, RouteIndexFileEntry } from '../clients/rewriter'
+import { currentDate } from '../resources/utils'
+
+const MAX_ROUTES_PER_REQUEST = 100
+const TEN_MINUTES_S = 10 * 60
+
+const sitemapIndexEntry = (forwardedHost: string, rootPath: string, entity: string, firstIndex?: number, lastIndex?: number): string => {
+  if (firstIndex && lastIndex) {
+    return `
+      <sitemap>
+        <loc>https://${forwardedHost}${rootPath}/newsitemap/${entity}-${firstIndex}-${lastIndex}.xml</loc>
+        <lastmod>${currentDate()}</lastmod>
+      </sitemap>
+    `
+  }
+  return `
+    <sitemap>
+      <loc>https://${forwardedHost}${rootPath}/newsitemap/${entity}.xml</loc>
+      <lastmod>${currentDate()}</lastmod>
```

Comment:
Can't we read this from rewritter indexing date? What is `endDate` value for?

Reply:
True. tks!